src/sprites/Elevator.lua

```diff
@@ -84,55 +101,79 @@ function Elevator:activate(key)
     -- Horizontal movement - get distance remaining
     movement = getMovementRemaining(self, key)
 
-    -- Update movement update vector applying orientation
-    self.movement = vector2D.new(movement, 0)
-elseif (key == KEYNAMES.Down or key == KEYNAMES.Up)
+  elseif (key == KEYNAMES.Down or key == KEYNAMES.Up)
   and self.fields.orientation == ORIENTATION.Vertical then
     -- Vertical movement - get distance remaining
     movement = getMovementRemaining(self, key)
-
-    -- Update movement update vector applying orientation
-    self.movement = vector2D.new(0, movement)
   end
-
+  
+  -- Set movement update scalar
+  self.movement = movement
+  
   -- Return boolean - did activation call capture movement
 
-  return movement ~= 0
+  return movement ~= 0, movement
 end
 
 function Elevator:update()
-  -- don't call SUPER update here, as that does collision work
+  Elevator.super.update(self)
+  
+  -- Don't call RigidBody update here, as that does collision work
```

Comment:
nit, but these commented out lines about RB can be removed, yeah

Reply:
Ah, right. Good catch there